tensorflow/lite/CMakeLists.txt

```diff
@@ -623,14 +642,27 @@ set(_ALL_TFLITE_SRCS
   ${TFLITE_SOURCE_DIR}/schema/schema_utils.cc
   ${TFLITE_SOURCE_DIR}/schema/schema_generated.h
 )
-add_library(tensorflow-lite
-  ${_ALL_TFLITE_SRCS}
+
+if (TFLITE_BUILD_SHARED_LIB)
+  #TODO: Add branch for windows (based on bazel - export all symbols)
+  if (APPLE)
+    target_link_options(tensorflow-lite PRIVATE "-Wl,-exported_symbols_list,${TENSORFLOW_SOURCE_DIR}/tensorflow/lite/tflite_exported_symbols.lds")
+  else ()
+    target_link_options(tensorflow-lite PRIVATE "-Wl,-z,defs")
+    target_link_options(tensorflow-lite PRIVATE "-Wl,--version-script,${TENSORFLOW_SOURCE_DIR}/tensorflow/lite/tflite_version_script.lds")
+  endif()
+endif()
+
+set_target_properties(tensorflow-lite
+  PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION  ${PROJECT_VERSION}
```

Comment:
How does this behave for static libraries?

Reply:
CMake applies the `VERSION` and `SOVERSION` properties only for shared library - https://cmake.org/cmake/help/latest/prop_tgt/VERSION.html , https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html .